GVFS/GVFS.UnitTests/Virtualization/Projection/SortedFolderEntriesTests.cs

```diff
@@ -116,6 +129,21 @@ public void ValidateOrderOfDefaultEntries()
             }
         }
 
+        [TestCase]
+        [Category(CategoryConstants.CaseSensitiveFileSystemOnly)]
+        public void ValidateCaseSensitiveOrderOfDefaultEntries()
+        {
+            List<string> allEntries = new List<string>(defaultFiles);
+            allEntries.AddRange(defaultFolders);
+            allEntries.Sort(CaseSensitiveStringCompare);
+            SortedFolderEntries sfe = SetupDefaultEntries();

```

Comment:
I'm wondering if we need to add some entries to the defaults that have different casing to make sure the sorting is happening correctly?  Like have a `Folder1` and `File1.txt`?  I think that could go in the default values because that would be good to have for the case insensitive tests as well.  Not sure if we would need the same entry that only differed in case?

Reply:
This turned out to be a little [more involved](https://github.com/microsoft/VFSForGit/pull/1412/commits/4435b1d03ed9a622fe80ca6ca4bfcdd53f0d61a3) than we might have hoped because [`GitIndexProjection.SortedFolderEntries.GetInsertionIndex()`](https://github.com/microsoft/VFSForGit/blob/9c733e5dd30b62d734eab6489d0410a17072b0c3/GVFS/GVFS.Virtualization/Projection/GitIndexProjection.SortedFolderEntries.cs#L165) does [platform-specific indexing](https://github.com/github/VFSForGit/blob/4435b1d03ed9a622fe80ca6ca4bfcdd53f0d61a3/GVFS/GVFS.Virtualization/Projection/GitIndexProjection.SortedFolderEntries.cs#L213) (as of this PR), so it throws an exception on Mac/Windows if their are case-differing-only entries in `defaultFiles` and `defaultFolders`.  Hence this more full-bodied approach.